src/HydrationMiddleware/HydratePublicProperties.php

```diff
@@ -36,9 +36,12 @@ public static function hydrate($instance, $request)
         $modelCollections = data_get($request, 'memo.dataMeta.modelCollections', []);
         $stringables = data_get($request, 'memo.dataMeta.stringables', []);
         $wireables = data_get($request, 'memo.dataMeta.wireables', []);
+        $customProperties = data_get($request, 'memo.dataMeta.customProperties', []);
 
         foreach ($publicProperties as $property => $value) {
-            if ($type = data_get($dates, $property)) {
+            if (in_array($property, $customProperties) && version_compare(PHP_VERSION, '7.4', '>=')) {
+                data_set($instance, $property, LivewireProperty::hydrate($instance, $property, $value));
+            } else if ($type = data_get($dates, $property)) {
```

Comment:
Maybe this method can be simplified to just use Reflection and defer to the property manager? Something like:

**A) The property has a declared type:**
1. Resolve the property type using Reflection
2. Check if the property manager has this type
    1. Yes — hydrate/dehydrate
    2. No — Check if the incoming value is `null` and whether the property supports `null`
        1. Yes — set as `null`
        2. No — Throw exception

**B) The property _does not_ have a declared type:**
1. Resolve the property type using the value's type
2. Go to A.2